pkg/network/ebpf/c/runtime/pid_mapper.c

```diff
@@ -0,0 +1,392 @@
+#include <linux/kconfig.h>
+#include "pid_mapper.h"
+#include <linux/net.h>
+#include <linux/fs.h>
+#include <linux/fdtable.h>
+#include <linux/sched.h>
+#include <net/sock.h>
+#include <linux/socket.h>
+
+#define SOCKET_INODE_FS_FN 1
+#define INET_STREAM_CONNECT 2
+
+// This map is used by unit tests to validate 
+// that the correct mapping was performed
+struct bpf_map_def SEC("maps/inode_pid_map") inode_pid_map = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(__u64),
+    .value_size = sizeof(int),
+    .max_entries = 0,
+    .pinning = 0,
+    .namespace = "",
+};
+
+struct bpf_map_def SEC("maps/save_pid") save_pid = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(__u64),
+    .value_size = sizeof(unsigned long),
+    .max_entries = 0, 
+    .pinning = 0,
+    .namespace = "",
+};
+
+struct bpf_map_def SEC("maps/symbol_table") symbol_table = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(__u64),
+    .value_size = sizeof(__u32),
+    .max_entries = 3,
+    .pinning = 0,
+    .namespace = "",
+};
+
+
+#define KERNEL_READ_FAIL(dest, sz, src)\
+    do {                                            \
+    if (bpf_probe_read_kernel(dest, sz, src) < 0)   \
```

Comment:
Do we want to retry for all error cases, or just something like `EINTR` or `EFAULT`?